### ISSUE TITLE
[Bug]: FTBChunks shows black map when switching servers behind Velocity Proxy

### ISSUE THREAD

**Issue by uberswe** [FTBTeam/FTB-Mods-Issues#1377]
### Mod

FTBChunks

### Mod version

ftb-chunks-fabric-2001.3.1

### Forge / Fabric version

1.20.10.15.11

### Modpack & version

Create: Gearworks 1.0.7

### What issue are you having?

Probably unsupported but also looking for advice. I am using a Velcity 3.3.0 Proxy to connect 9 different fabric servers which all have the same modpack which includes FTBChunks. When switching between the servers users get a black map until they change dimension. After changing dimensions the map works until they switch to a new server again.

### Crashlogs

no crash

### Steps to reproduce

1. Setup a Velcity Proxy
2. Setup 2 fabric servers wtih FTBChunks, Crossstitch and FabricProxy-Lite
3. Join the proxy and one of the servers, use /server server2 to switch servers
4. See black map when opening FTBChunks map

### Anything else to note?

I am wondering if I could trigger the refresh via a mixin or something similar? My Java skills aren't excellent but with some guidance/hints I might be able to fix this with my own mod as I am not sure you will fix this issue.

This is my current fix for the issue, on the server I use the following code when a player joins to change their dimension.

```
ServerPlayConnectionEvents.JOIN.register(((handler, sender, server) ->
                PlayerDataService.fakeDimensionSwap(handler.getPlayer(), server, config)));
```

```
    public static void fakeDimensionSwap(ServerPlayerEntity player, MinecraftServer server, Config config) {
        Logger.log(Level.INFO, "fake dimension swap start");
        ServerWorld currentWorld = (ServerWorld) player.getWorld();
        RegistryKey<World> fakeDimension = World.NETHER;
        if (currentWorld.getDimensionEntry() == World.NETHER.getValue()) {
            fakeDimension = World.END;
        }
        ServerWorld fakeWorld = server.getWorld(fakeDimension);
        Vec3d originalPos = player.getPos();
        TickScheduler.schedule(() -> {
            // Send the packet to the client
            FabricDimensions.teleport(player, fakeWorld, new TeleportTarget(Vec3d.ZERO.add(128, 128, 128), Vec3d.ZERO, player.getYaw(), player.getPitch()));
        }, 10);
        TickScheduler.schedule(() -> {
            FabricDimensions.teleport(player, currentWorld, new TeleportTarget(originalPos, Vec3d.ZERO, player.getYaw(), player.getPitch()));
            Logger.log(Level.INFO, "fake dimension swap end");
        }, 10 + config.serverHopDelayTicks);
    }
```

**Comment by uberswe** [comment]
I think I'm able to fix this by calling shutdown when joining another server, so basically if MapManager instance is already running we should shut it down before starting up a new one.

```
package com.gearworkssmp.gearworks.mixin;

import com.gearworkssmp.gearworks.Gearworks;

import dev.ftb.mods.ftbchunks.client.map.MapManager;

import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.Pseudo;
import org.spongepowered.asm.mixin.Unique;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

import java.util.UUID;


@Pseudo
@Mixin(MapManager.class)
public class FTBChunksMapManagerMixin {

	@Inject(method = "startUp", at = @At("HEAD"), remap = false)
	private static void onStartUp(UUID serverId, CallbackInfo ci) {
		Gearworks.LOGGER.info("onStartupCalled: {}", serverId);
		if (MapManager.getInstance().isPresent()) {
			Gearworks.LOGGER.info("Shutting down before starting: {}", MapManager.getInstance().get().getServerId());
		}
		additionalShutDown(serverId);
	}

	@Unique
	private static void additionalShutDown(UUID serverId) {
		Gearworks.LOGGER.info("server id, shutdown called: {}", serverId);
		MapManager.shutdown();
	}
}
```

**Comment by desht** [comment]
I can incorporate this change (which looks fine to me) into `MapManager`, but please don't publish a mod with a mixin to one of our mods. There's every chance a future change will lead to a crash.

**Comment by uberswe** [comment]
> I can incorporate this change (which looks fine to me) into `MapManager`, but please don't publish a mod with a mixin to one of our mods. There's every chance a future change will lead to a crash.

Thanks, I will keep that in mind. This mod is mainly only used for my own server and I wasn't expecting you to fix such an edge case. I appreciate the quick response and will remove this as soon as the fix is live and try to come to you before doing any mixins in the future.

**Comment by desht** [comment]
Should be fixed in FTB Chunks 2101.1.3

**Comment by MineYF** [comment]
> 应该在 FTB Chunks 2101.1.3 中修复

hello,Can this fix be applied to the 1.20.1 fabric? I'm also experiencing this issue and would like to have it resolved on the 1.20.1 version. Thank you!
